docs/api-docs/in-depth-guides/rail-integration-guide.mdx

```diff
@@ -1,100 +1,154 @@
+    Z[Rail Interchange Received] --> B
     A[Rail Loaded] --> B[Rail Departed]
+    B --> L[Train Passing]
     B --> C[Rail Arrived]
+    L --> L
+    L --> C
     C --> D[Rail Unloaded]
-    D --> E[Full Out]
+    C --> X[Rail Interchange Delivered]
+    X --> Z
+    D --> G[Available for Pickup]
+    D --> H[Not Available]
+    H -- Holds and Fees Updated --> H
+    G --> H
+    H --> G
+    G --> E[Full Out]
     E --> F[Empty Return]
-  ```
+```
 
-```mermaid
-timeline
-    title Container Milestones (Rail Import)
-    section Port of Discharge
-        Vessel Arrival : Port of discharge : Terminal details : discharged timestamp
-        Vessel Birth : Port of discharge : Terminal details : Birth timestamp
-        Container Discharged : Port of discharge : Terminal details : discharged timestamp
-    section Container On Rail
-        Rail Loaded : Loaded location : loaded timestamp
-        Rail Departed : Departed location : departed timestamp
-        Rail Arrived : Arrived location : arrived timestamp
-        Rail Unloaded : Unloaded location : unloaded timestamp
-    section Final Destination (Rail Terminal)
-        Full Out : Full out port : Full out terminal : Full out timestamp
-        Empty Return : empty return timestamp
+### Webhook Notifications
 
-```
+Terminal49 provides webhook notifications to keep you updated on key milestones in a container's rail journey. These notifications allow you to integrate near real-time tracking data directly into your applications.
+
+Here's a list of the rail-specific events which support webhook notifications:
+
+| Milestone | Webhook Notification | Description |
+|----------------|----------------------|-------------|
+| Rail Loaded    | `container.transport.rail_loaded`   | The container is loaded onto a railcar. |
+| Rail Departed  | `container.transport.rail_departed` | The container departs on the railcar (usually from port of discharge). |
+| Rail Arrived   | `container.transport.rail_arrived`  | The container arrives at the destination terminal. |
+| Rail Unloaded  | `container.transport.rail_unloaded` | The container is unloaded from a railcar. |
+
+There's also a set of events that are triggered when the status of the container at the destination rail terminal changes.  For containers without rail, they would have been triggered at the ocean terminal.
 
-#### Rail Milestones (aka rail events)
+| Milestone | Webhook Notification | Description |
+|----------------|----------------------|-------------|
+| Available for Pickup | `container.transport.available`     | (Coming Soon) The container is available for pickup at the rail terminal. |
+| Not Available  | `container.transport.not_available` | (Coming Soon) The container is not available for pickup at the rail terminal. |
+| Holds and Fees Updated | `container.transport.holds_and_fees_updated` | (Coming Soon) The holds and fees for the container at the rail terminal have been changed, but not fully cleared. |
```

Comment:
If they're all removed then they may have cleared. we actually don't know with certainty the have so I would omit _, but not fully cleared_

Reply:
Omitting "but not fully cleared" would change the meaning of the event.  It might be an improvement, might not be.

Originally, I intended for only one of the following events to trigger at a time:
* Available
* Not Available
* Holds and Fees Changed

So "Holds and Fees Changed" would only be triggered when a hold or fee was changed, but there were still some on the container, so it was still Not Available.  If all were removed, then the Available event would fire.

If we remove that restriction, then we would get both "Available" and "Holds and Fees Changed" events when removing the last hold or fee.  We'd also get both "Not Available" and "Holds and Fees Changed" when adding the first hold or fee.

Is that what we want?